### ISSUE TITLE
`assertRejects` and `assertThrows` should return the caught error

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_std#2220]
It'd be nice if these functions returned the error they caught, so users could do more intricate assertions on the error itself.

**Comment by bartlomieju** [comment]
Sounds good to me, this should be a simple change.

**Comment by mrkldshv** [comment]
Hey! I'd like to take this issue.

**Comment by bartlomieju** [comment]
@mrkldshv please do!

**Comment by kt3k** [comment]
`assertThrows` accepts `errorCallback` as the 2nd arg and you can assert on details about the thrown error in it.

I think we discussed similar suggestion before, but we didn't accept it based on the design principle `assertion functions shouldn't return anything` ref: https://github.com/denoland/deno_std/pull/1052#discussion_r674510375

**Comment by kt3k** [comment]
(Note: we added `errorCallback` overload in https://github.com/denoland/deno_std/pull/1219)

**Comment by mrkldshv** [comment]
Right, I've just looked through previous discussion. As pointed by @kt3k, it wasn't added because `assertion functions shouldn't return anything`. What do you think we should do? @lucacasonato @bartlomieju

**Comment by lucacasonato** [comment]
I think the "Assertions shouldn't have return values." rule is rather arbitrary. I think in this case returning an error is totally reasonable. the error callback does not really solve the issue for me, because I may need to do some async work with the error

**Comment by KyleJune** [comment]
In this previous issue I proposed a similar change to having these functions return the values thrown. I believe the fix for this could also be made to be a fix for that issue. Here is a link to my comment there. It would be good to keep error class and message arguments as a shorthand for both asserting that they throw/reject and asserting that it is an Error that was thrown.

https://github.com/denoland/deno_std/issues/1362#issuecomment-940101918

**Comment by kt3k** [comment]
> I think the "Assertions shouldn't have return values." rule is rather arbitrary. I think in this case returning an error is totally reasonable. the error callback does not really solve the issue for me, because I may need to do some async work with the error

I'm open to changing that rule now as it's requested/suggested many times and it seems to have some practical use case.

@kitsonk What do you think about this suggested change? Do you have any concerns/objections?

**Comment by mrkldshv** [comment]
@kt3k Should we close this issue?

**Comment by kt3k** [comment]
@mrkldshv Yes, thanks!
